src/main/java/org/alfresco/repo/jscript/RhinoScriptProcessor.java

```diff
@@ -453,14 +456,16 @@ private Object executeScriptImpl(Script script, Map<String, Object> model, boole
         {
             // Create a thread-specific scope from one of the shared scopes.
             // See http://www.mozilla.org/rhino/scopes.html
-            cx.setWrapFactory(wrapFactory);
+            
             Scriptable scope;
             if (this.shareSealedScopes)
             {
-                Scriptable sharedScope = secure ? this.nonSecureScope : this.secureScope;
+                Scriptable sharedScope = secure ? this.nonSecureScope : this.secureScope;
+                cx.setWrapFactory(secure ? wrapFactory : sandboxFactory);

```

Comment:
1. Why has this line been moved from outside of the if(this.shareSealedScops) block into this if block?
2. If this.shareSealedScopes is false, then cx.setWrapFactory() does not get invoked